src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -155,32 +182,92 @@ public String formatRequest(Position position) throws UnsupportedEncodingExcepti
     @Override
     protected Position handlePosition(Position position) {
 
-        String url;
-        if (json) {
-            url = this.url;
-        } else {
-            try {
-                url = formatRequest(position);
-            } catch (UnsupportedEncodingException | JsonProcessingException e) {
-                throw new RuntimeException("Forwarding formatting error", e);
+        class AsyncRequestAndCallback implements InvocationCallback<Response> {
```

Comment:
Why is the class defined inline in a method? I would recommend to move it out to the class scope.

Reply:
Would you like to move it before `WebDataHandler`'s constructor or just before `handlePosition` method?

Reply:
What about this?

Reply:
What about this?

Reply:
Would you like to move `AsyncRequestAndCallback` class implementation before `WebDataHandler`'s constructor, just bellow member variables declaration, or just before `handlePosition` method? Or anywhere else? Thanks.

Reply:
I would move it above `handlePosition` or below everything.

Reply:
Ok. Thanks.